lib/src/common/cache/most_recently_used_strategy.dart

```diff
@@ -0,0 +1,49 @@
+import 'dart:async';
+import 'dart:collection';
+
+import 'package:w_common/cache.dart';
+
+/// A [CachingStrategy] that will hold the last n most recently used [TValue]s.
```

Comment:
I think there might be the potential for some ambiguity/misunderstanding with this cache type.  Is this a ‘least recently used’ cache?

>  Discards the least recently used items first.
[https://en.m.wikipedia.org/wiki/Cache_replacement_policies#Least_Recently_Used_(LRU)](https://en.m.wikipedia.org/wiki/Cache_replacement_policies#Least_Recently_Used_(LRU))

If so, I think we should name this strategy closer to what the generally accepted cache type is believed to be.